platform-api/che-core-api-account/src/main/java/org/eclipse/che/api/account/server/AccountService.java

```diff
@@ -580,14 +588,49 @@ public void remove(@ApiParam(value = "Account ID", required = true)
         accountDao.remove(id);
     }
 
+    @POST
+    @Path("/{accountId}/{workspaceId}")
+    @Produces(APPLICATION_JSON)
+    @RolesAllowed("account/owner")
+    public AccountDescriptor registerWorkspace(@PathParam("accountId") String accountId, @PathParam("workspaceId") String workspaceId)
+            throws NotFoundException, ServerException, BadRequestException, ConflictException {
+        Account account = accountDao.getById(accountId);
+        UsersWorkspace workspace = workspaceManager.getWorkspace(workspaceId);
+        if (accountDao.isWorkspaceRegistered(workspaceId)) {
```

Comment:
The concept is close to [this one](https://github.com/codenvy/che-core/blob/4.0/platform-api/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceService.java#L244-L256).
I think `PUT` may be used in this case, but `POST` looks okay too. Maybe guys have something to say about it @skabashnyuk, @akorneta ?

The workspace relation to account is _many to one_, it is important because in the other way it would not be possible to detect the account which should pay for used resources. Or what do you mean by multiple times?